src/library_fs.js

```diff
@@ -1109,10 +1109,9 @@ mergeInto(LibraryManager.library, {
       if (!stream.stream_ops.read) {
         throw new FS.ErrnoError(ERRNO_CODES.EINVAL);
       }
-      var seeking = true;
-      if (typeof position === 'undefined') {
+      var seeking = position != null;
```

Comment:
`!=` is a less strict comparison than `!==`. What exactly is being checked here - if it's undefined or null? Or if it is any falsey value?

Reply:
It's checking nullish value (`null` and `undefined`). Effectively same as `typeof position === 'undefined' || position === null)` but more compact version.

Reply:
I see, thanks. I think this makes the code a little harder to read, though - for example I didn't remember that `x != null` is true if and only if x is null or undefined. How about requiring this to be either a number or null, or either a number or undefined? (Another option could have been to check `!position` but that would give true for 0, which is valid here.)

Reply:
Maybe a comment will be sufficient, but I think forcing `undefined` is okay for now. Someone who want to allow `null` can open another PR 👍